test/e2e/client_test.go

```diff
@@ -178,3 +187,34 @@ func (t *testClient) dumpInfo() error {
 	}
 	return nil
 }
+
+func (t *testClient) instantiateTemplate(tpl string) error {
+	// Create the template
+	template, err := t.tc.Templates("openshift").Get(
+		tpl, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	// Instantiate the template
+	_, err = t.tc.TemplateInstances(c.namespace).Create(
+		&templatev1.TemplateInstance{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: c.namespace,
+			},
+			Spec: templatev1.TemplateInstanceSpec{
+				Template: *template,
+			},
+		})
+	if err != nil {
+		return err
+	}
+
+	// Poll for the deployment status
+	err = wait.PollImmediate(2*time.Second, 10*time.Minute, c.templateInstanceIsReady)
```

Comment:
nit: you can shortcut everything below to `return wait.PollImmediate(2*time.Second, 10*time.Minute, c.templateInstanceIsReady)`